src/main/java/com/bobocode/hoverla/bring/context/BeanConfigurationClassScanner.java

```diff
@@ -0,0 +1,91 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.annotation.Bean;
+import com.bobocode.hoverla.bring.annotation.Configuration;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.reflections.Reflections;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A bean definition {@link BeanScanner scanner} that detects bean candidates defined as methods marked
+ * with {@link Bean @Bean} annotation in class annotated with {@link Configuration @Configuration}
+ *
+ * @see Bean @Bean
+ * @see Configuration @Configuration
+ * @see BeanScanner
+ */
+@Slf4j
+public class BeanConfigurationClassScanner implements BeanScanner {
+
+    private final String[] packagesToScan;
+    private final BeanConfigClassValidator validator;
+    private final BeanDefinitionMapper mapper;
+
+    public BeanConfigurationClassScanner(BeanConfigClassValidator validator,
+                                         BeanDefinitionMapper mapper,
+                                         String... packagesToScan) {
+        this.packagesToScan = packagesToScan;
+        this.validator = validator;
+        this.mapper = mapper;
+    }
+
+    /**
+     * Performs scan of classes annotated with {@link Configuration @Configuration}
+     * from given {@link BeanConfigurationClassScanner#packagesToScan packages}.
+     *
+     * <p>Scanned {@link Class} objects are then transferred to {@link BeanConfigurationClassScanner} for further validation process.</p>
```

Comment:
I guess it should be link to BeanConfigClassValidator or BeanConfigurationClassScanner.scanBeanDefinition().

Reply:
good catch

Reply:
fixed